### ISSUE TITLE
Remove `foo$` style names in sync documentation

### ISSUE THREAD

**Issue by ben-albrecht** [chapel-lang/chapel#19152]
Now that implicit read/writes of syncs are deprecated (https://github.com/chapel-lang/chapel/pull/17287), it is no longer suggested to annotate syncs by appending `$` to their variable name. However, our documentation on syncs still uses this naming style:

- [sync primer](https://chapel-lang.org/docs/1.25/primers/syncsingle.html)
- [sync spec](https://chapel-lang.org/docs/1.25/language/spec/task-parallelism-and-synchronization.html?highlight=sync)

Should we update these to remove the `foo$` naming style of sync variables?

**Comment by Guillaume-Helbecque** [comment]
Hello, I can work on this issue if you agree and if nobody jumped on it already.

**Comment by bradcray** [comment]
@Guillaume-Helbecque : I don't believe anyone has, and would be happy to see this happen, but also don't want to rely on you to clean up our messes for us if you have more important things to work on.

**Comment by Guillaume-Helbecque** [comment]
@bradcray It's a pleasure for me to help with the project at my level. This is something I can work on the rainy days :)

**Comment by Guillaume-Helbecque** [comment]
I just have two questions regarding the PR:
- It seems that the only reason why `$` has been authorized in the language is for the convention, right? Whereas I think you should keep `$` authorized, how to justify it now?
https://github.com/chapel-lang/chapel/blob/2bfefa51d6468a6b111ae7bcd84d4ef9b5fd2b79/doc/rst/language/spec/lexical-structure.rst#L112-L117
- Should we just remove the suggestion, or leave it but as a "deprecated suggestion"? In order to facilitate the transition between both.
https://github.com/chapel-lang/chapel/blob/2bfefa51d6468a6b111ae7bcd84d4ef9b5fd2b79/doc/rst/language/spec/task-parallelism-and-synchronization.rst#L238-L240

**Comment by bradcray** [comment]
@Guillaume-Helbecque : Good questions!

For the first: I'd forgotten that text existed in the spec.  For the sake of making progress on this issue, I'd probably just drop this rationale and not bother trying to rationalize the choice of permitting identifiers to include `$` (I view the existing text as part-rationale, part trying to indoctrinate people into the old way of doing things early).

If you felt so inclined, you could fork off a new issue asking whether we wanted to continue permitting `$` in identifier names (or make it unstable for 2.0).  If you don't, I may do that since you've raised the question.

> Should we just remove the suggestion, or leave it but as a "deprecated suggestion"? In order to facilitate the transition between both.

In the spec, I would just remove the suggestion.  We usually try to make the spec reflect the language as we envision it as accurately as possible, and don't typically note changes with deprecations as we do in the library documentation.
